cmd/opm/registry/serve.go

```diff
@@ -66,44 +79,76 @@ func serveFunc(cmd *cobra.Command, args []string) error {
 		logrus.WithError(err).Warn("unable to write default nsswitch config")
 	}
 
-	dbName, err := cmd.Flags().GetString("database")
-	if err != nil {
-		return err
-	}
-
 	port, err := cmd.Flags().GetString("port")
 	if err != nil {
 		return err
 	}
 
-	logger := logrus.WithFields(logrus.Fields{"database": dbName, "port": port})
+	logger := logrus.WithField("port", port)
 
-	// make a writable copy of the db for migrations
-	tmpdb, err := tmp.CopyTmpDB(dbName)
+	source, mode, err := detectRegistrySource(cmd, args)
 	if err != nil {
 		return err
 	}
-	defer os.Remove(tmpdb)
 
-	db, err := sqlite.Open(tmpdb)
-	if err != nil {
-		return err
-	}
+	var store registry.GRPCQuery
 
-	// migrate to the latest version
-	if err := migrate(cmd, db); err != nil {
-		logger.WithError(err).Warnf("couldn't migrate db")
-	}
+	switch mode {
+	case modeDeclCfg:
+		logger = logger.WithField("configs", source)
 
-	store := sqlite.NewSQLLiteQuerierFromDb(db)
+		var (
+			cfg *declcfg.DeclarativeConfig
+			err error
+		)
+		if s, sErr := os.Stat(source); sErr != nil {
+			return err
+		} else if s.IsDir() {
+			cfg, err = declcfg.LoadDir(source)
+		} else {
+			cfg, err = declcfg.LoadFile(source)
+		}
```

Comment:
Another minor nit: this likely can be reduced to an if/else given you return early in while stat-ing the source directory.

Reply:
Now I'm re-reading that section of code and I'm trying to figure out if I mistakenly used `err` when I should have used `sErr` if `os.Stat` returns an error.

Reply:
^ Fixed